base/cvd/cuttlefish/host/commands/cvdalloc/cvdalloc.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <string.h>
+#include <unistd.h>
+
+#include <android-base/logging.h>
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/strings/str_format.h"
+
+#include "allocd/alloc_utils.h"
+#include "cuttlefish/common/libs/fs/shared_fd.h"
+
+ABSL_FLAG(int, id, 0, "Id");
+ABSL_FLAG(int, teardown_event_socket, 0, "Socket");
+ABSL_FLAG(int, allocate_event_socket, 0, "Socket");
```

Comment:
Do these need to be separate sockets? Presumably if `run_cvd` closes `allocate_event_socket` it also isn't interested in using the resources.